File: src/leetcode/hash_table/T128.rs
use std::collections::HashSet;

pub struct Solution {}

impl Solution {
    pub fn longest_consecutive(nums: Vec<i32>) -> i32 {
        let mut map = HashSet::new();
        for num in &nums {
            map.insert(num);
        }
        let mut res = 0;

        for num in &nums {
            let mut temp = num.clone();
            let mut count = 1;
            if !map.contains(&temp) {
                continue;
            }
            while map.len() > 0 && map.contains(&((temp + 1).clone())) {
                map.remove(&((temp + 1).clone()));
                count += 1;
                temp += 1;
            }
            temp = num.clone();
            while map.len() > 0 && map.contains(&((temp - 1).clone())) {
                map.remove(&((temp - 1).clone()));
                count += 1;
                temp -= 1;
            }
            if count > res {
                res = count;
            }

        }
        res
    }
}

File: src/main/java/org/example/array/T128.java
package org.example.array;

import java.util.HashSet;
import java.util.Set;

/**
 *
 * @Datetime: 5/1/2024 下午3:17
 * @Author: koopkl
 * @Link: <a href="https://leetcode.cn/problems/longest-consecutive-sequence/description/">最长连续序列</a>
 */
public class T128 {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for (int j : nums) {
            set.add(j);
        }
        int res = 0;
        for (Integer num : set) {
            int tempRes = 1;
            if (!set.contains(num - 1)) {
                int nowNum = num;
                while (set.contains(nowNum + 1)) {
                    nowNum += 1;
                    tempRes += 1;
                }
            }
            res = Math.max(res, tempRes);
        }
        return res;
    }

    public static void main(String[] args) {
        System.out.println(new T128().longestConsecutive(new int[]{100, 1, 2, 3, 200}));
    }
}
